tests/integration/kubernetes/confidential_common.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bats
```

Comment:
Let's try to keep the same files hierarch as in the CCv0 branch, that will help us on the migration later. On CCv0 we have:
```
[wmoschet@wainer-laptop kubernetes]$ tree -d
.
├── confidential
│   └── fixtures
├── e2e_conformance
├── kubeadm
├── runtimeclass_workloads
│   └── stress
└── volume

8 directories
[wmoschet@wainer-laptop confidential]$ tree
.
├── agent_image.bats
├── agent_image_encrypted.bats
├── fixtures
│   ├── pod-config.yaml.in
│   └── service.yaml.in
├── lib.sh
├── sev.bats
└── snp.bats

2 directories, 7 files
```

The conventions that we took are more or less those:

1. If utility code shared amongst k8s tests, put on `confidential/lib.sh`
2. Static files and/or templates used by the k8s tests, put on `confidential/fixtures`
3. k8s bats tests are loosely on `confidential`
4. Any TEE specific tests goes to <tee>.bats on `confidential` but we want to avoid TEE-specific tests
5. Common test code which are not k8s specific (e.g. tweaks to containerd configuration) goes two level up, on `tests/confidential/lib.sh` (same idea for fixtures, they are put on `tests/confidential/fixtures`).

Reply:
So, I see the suggestion of using fixtures here, coming up again.

I'm **NOT** against the idea, but I'm against the idea of having only part of the tests using fixtures.
With this in mind, I've privately spoken with @wainersm who agreed to port all the current tests to use fixtures.

I'd wait till that happens before proceeding with this one, so we can make sure we have **ONE WAY** of doing things.

Reply:
Can we name the actual helper scripts something more descriptive than `lib.sh`? I know they have a descriptive parent folder, but sometimes scripts can float around.

Another issue that we had in the tests repo, is that the lib scripts required to be in a specific location in the repo and looked for other things in the repo. The more standalone a function library script can be, the better IMHO. I suppose we can just try to enforce this in code review.

I think it would help to have a discussion on how many helper scripts, what they should be named, and where they should be located. One adverse thing to sticking them in sub folders all over the repo is that it makes them hard to find, and then someone comes along and adds another one somewhere else. Would it be better to have all helper scripts located in one directory? One library script or many?

Was thinking some potential library script names:
```
lib-docker.sh
lib-k8s.sh
lib-kata.sh
```

Maybe I should have posted this conversation in the CI channel. Let me know if you guys want me to move it there.

Reply:
@wainersm I think in my opinion adding another directory layer might complicate things further. 
1. Utility code will be placed in `confidential_common.sh`
2. We just need yaml as a static file that is placed in runtimeclass_workloads
3 & 4: No TEE specific tests, only generic tests. And those generic tests will be placed in in `tests/integration/kubernetes` dir along with all other bats files.
5: I don't think we have a common code that is not related to k8s.

Please let me know what you think @wainersm .